source/common/network/connection_impl.cc

```diff
@@ -813,6 +816,9 @@ void ServerConnectionImpl::raiseEvent(ConnectionEvent event) {
 void ServerConnectionImpl::onTransportSocketConnectTimeout() {
   stream_info_.setConnectionTerminationDetails(kTransportSocketConnectTimeoutTerminationDetails);
   closeConnectionImmediately();
+  if (transport_socket_timeout_stat_ != nullptr) {
```

Comment:
When this timeout callback is invoked, is nullptr only possible in test cases?

Reply:
Yes, I think so; but if there are other use cases I'm not aware of, I didn't want it to crash.

Reply:
I don't think it will be ever possible to be a nullptr, from where this function is called. The parameter should be a reference.